tabcmd/commands/auth/session.py

```diff
@@ -193,14 +169,52 @@ def check_last_login_username_token_name(self):
                 last_login_token_name_present, \
                 username, token_name
 
-    def create_new_session_using_username(self, args):
-        self.update_session(args)
-        signed_in_object \
-            = self.no_cookie_save_session_creation_with_username()
-        return signed_in_object
+# json file functions
 
-    def create_new_session_using_token(self, args):
-        self.update_session(args)
-        signed_in_object \
-            = self.no_cookie_save_session_creation_with_token()
-        return signed_in_object
+    def _get_file_path(self):
+        home_path = os.path.expanduser("~")
```

Comment:
I can see a possible issue happen here when we create an executable from this code. If users have to run as `sudo` , `home_path` will be `/root/` and not home. This should either be hardcoded further to be 
Maybe we should just add a note when we create/test executable to not run as `sudo`  or add some code like

```
username = os.getenv("USER") 
if(username != 'root'):
    home_path=os.path.expanduser('~'+username)
else:
   home_path = os.path.expanduser("~")
```

I'm not sure how the above block will work in windows, I haven't tried it.

Reply:
https://github.com/tableau/tabcmd/issues/45